modules/tinymce/src/themes/silver/main/ts/ui/context/ContextFormApi.ts

```diff
@@ -7,17 +7,17 @@ import {
 } from '@ephox/alloy';
 import { InlineContent } from '@ephox/bridge';
 import { Singleton } from '@ephox/katamari';
+import { Focus } from '@ephox/sugar';
 
-import * as ContextToolbarFocus from './ContextToolbarFocus';
 import { backSlideEvent } from './ContextUi';
 
-export const getFormApi = <T>(input: AlloyComponent): InlineContent.ContextFormInstanceApi<T> => {
+export const getFormApi = <T>(input: AlloyComponent, focusfallbackComp?: AlloyComponent): InlineContent.ContextFormInstanceApi<T> => {
```

Comment:
Since all we need is the element, not the whole component, I suggest reducing the coupling here. That would also reduce the confusing way you pass in both the result of `memInput.get(comp)` as well as `comp` itself. That's somewhat Alloy's fault, but I think we can still be clearer about what the intent is here.

tl;dr pass in a function that returns the fallback element instead of a "fallback component".

Reply:
changed